### PR TITLE
Pagination Init

### PR DESCRIPTION
**Pull request by tuna42na**
This addition adds pagination as well as a styled button trial to the list page.

### CODE REVIEW

File: src/components/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  });
+
+  // Input Field Handling
+  const handleChange = (e) => {
+    setNewUser({ ...newUser, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    addNewUser(newUser);
```
Comment (trvslhlt) [inline src/components/SignUp.js:25]: What happens if there is an error when trying to add a new user?

File: src/contexts/NameListContext.js [hunk 2]
```diff
@@ -125,6 +128,17 @@ class NameListProvider extends React.Component {
     this.setState({ namesPerPage: number });
     let results = [...this.state.list];
     this.setState({ listView: results.splice(0, number) });
+    this.onPageChange(this.state.page);
+  };
+
+  // Change Page
+  onPageChange = (newPageNumber) => {
+    let pageStart = (newPageNumber - 1) * this.state.namesPerPage;
```
Comment (trvslhlt) [inline src/contexts/NameListContext.js:128]: How does this function behave if `newPageNumber` is too small or too large?

File: src/contexts/UserContext.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    this.state = {
+      // Functional Exports
+      addNewUser: this.addNewUser,
+      deleteUser: this.deleteUser,
+    };
+  }
+
+  // Add and Delete Actions
+  addNewUser = (userData) => {
+    axios.post(USER_URL, { userData }).then((res) => {
```
Comment (trvslhlt) [inline src/contexts/UserContext.js]: It looks like `axios.post` returns a promise. If so then that needs to be handled with a catch or returned to the caller to let them handle it.

**Review by trvslhlt — CHANGES_REQUESTED** [review]
I may have some more comments, but here are the first few things that caught my attention.

File: src/components/SignUp.js (hunk 1 above)
Comment (tuna42na) [reply src/components/SignUp.js:25]: @travis yea. I have yet to actually hookup the user addition side of things. These files in the PR a really just the current state of things but most of the work on this I've done for the pagination.

File: src/contexts/NameListContext.js (hunk 2 above)
Comment (tuna42na) [reply src/contexts/NameListContext.js:128]: I have it to conditionally render the previous button to not appear on the first page. Just added a new function to not go over, but gonna try and make it look more elegant...

**Review by tuna42na — COMMENTED** [review]
Yup

**Pull request merged**